frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_statistics_modal.tsx

```diff
@@ -190,17 +233,38 @@ export function SegmentStatisticsModal({
     return potentialGroupNode?.name == null ? "" : potentialGroupNode.name;
   };
 
+  const isErrorCase = dataSource === ERROR_CODE;
```

Comment:
instead of this variable, you can now use `dataSource != null` everywhere directly. then, TS should also understand that dataSource will be an array (and not a string). so, the casting won't be necessary anymore.